__tests__/environment-test.js

```diff
@@ -0,0 +1,81 @@
+"use strict"
+
+jest.dontMock('../environment.js');
+
+describe('environment', () => {
+  let environment = require('../environment.js');
+
+  it('uses original config attributes if no env key is found', () => {
+    let parsedConfig = environment.checkEnvironmentForConfig({
+      key:'fromDefaultConfig'
+    })
+
+    expect(parsedConfig).toEqual({
+      key:'fromDefaultConfig'
+    });
+  })
+
+  it('correctly interperts the current config', () => {
+    process.env.MENTION_BOT_CONFIG = `
+      {
```

Comment:
When the next person is going to want to add another option, s/he is going to grep the project for an existing one and see 3 lines in this file, that do not really matter.

I would recommend to test cases such as number, array, boolean, string with random names instead of the full config.